solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -762,7 +761,7 @@ private boolean returnStoredFields() {
     }
 
     private boolean returnDVFields() {
-      return CollectionUtils.isNotEmpty(dvFields);
+      return !(dvFields != null && dvFields.isEmpty());
```

Comment:
A comment in line 738 says `dvFields` is always not null, not can simplify
```suggestion
      return !dvFields.isEmpty();
```